hazelcast/connection.py

```diff
@@ -146,17 +146,20 @@ def add_listener(self, on_connection_opened=None, on_connection_closed=None):
     def get_connection(self, member_uuid):
         return self.active_connections.get(member_uuid, None)
 
-    def get_random_connection(self):
+    def get_random_connection(self, should_get_data_member=False):
         if self._smart_routing_enabled:
-            member = self._load_balancer.next()
-            if member:
-                connection = self.get_connection(member.uuid)
-                if connection:
-                    return connection
+            connection = self._get_connection_from_load_balancer(should_get_data_member)
+            if connection:
+                return connection
 
         # We should not get to this point under normal circumstances.
```

Comment:
Yes, let me update it, the comment only makes sense for a smart client. The copying is done to not get concurrent modification exceptions since that dict might be modified by reactor thread while iterating.